### PR TITLE
Delete any existing gdb_cmds file before writing new contents.

### PR DESCRIPTION
**Pull request by donmr**
run_stm32_prog needs to delete any existing gdb_cmds file before it adds new content.

Can you please merge this before making large changes to master so it doesn't suffer the same issues that the other branch did?

### CODE REVIEW

**Comment by drowe67** [comment]
@donmr yes sorry about the mess I made on your other branch.  

When I try to run a test using this PR I'm getting:
```
rm: cannot remove 'gdb_cmds': No such file or directory
```
and the test fails ....

I wasn't sure if the test_run/test_name dir was automatically cleaned out on every run?  Have looked at the scripts but can't work out where the dir for the test is created.

File: stm32/unittest/scripts/run_stm32_prog [hunk 1]
```diff
@@ -28,8 +28,9 @@ else
   UT_SH_FIO="enable"
 fi
 
-if [ ! ${ARGS[--st-util]+_} ] ; then
-   echo "----------------------- STARTING gdb/OpenOCD ----------------------------"
+rm gdb_cmds
```
Comment (db4ple) [inline stm32/unittest/scripts/run_stm32_prog]: Make it 'rm -f gdb_cmds'

**Comment by donmr** [comment]
Yes, I should have run more tests...

**Comment by drowe67** [comment]
> run_stm32_prog needs to delete any existing gdb_cmds file before it adds new content.
> 
> Can you please merge this before making large changes to master so it doesn't suffer the same issues that the other branch did?

@donmr I'd like to merge https://github.com/drowe67/codec2/pull/41, which is a large change to master, although just minor changes to the stm32 side.

You've asked me to hold off on large changes to master until this PR is merged but it isn't passing the tests.  Can you pls fix and test this PR ASAP so I can merge this branch first as requested?

**Comment by donmr** [comment]
OK, I added the "-f" to avoid the message when the file is not there.
However that is not why the test failed.  It failed because this script was still looking to see if there was a "--openocd" option even though that is now the default.  And it fails with st-util.  I changed the check to look for "--st-util" instead.  I thought this had already been done by someone else when the other changes were made.  Perhaps that is in some other branch???  Tests now pass for me.

**Comment by drowe67** [comment]
Thanks Don.  I get 23/24 test pass, the one that failed:
```
Start 22: tst_api_demod_700D_AWGN_codec
<snip>
22: ref energy: 107.823803 target energy: 106.709207
22: energy compare BAD
```
The energy test is my way of comparing the speech waveforms. The threshold is 1dB. But that's OK for now, I'll merge it, and will sort that fail out if it reappears when https://github.com/drowe67/codec2/pull/41 is merged (it passes in https://github.com/drowe67/codec2/pull/41).

**Comment by drowe67** [comment]
> OK, I added the "-f" to avoid the message when the file is not there.
> However that is not why the test failed. It failed because this script was still looking to see if there was a "--openocd" option even though that is now the default. And it fails with st-util. I changed the check to look for "--st-util" instead. I thought this had already been done by someone else when the other changes were made. Perhaps that is in some other branch??? Tests now pass for me.

Yes I thought I had fixed that some time ago.  Perhaps something off happened with the merge, or I messed up with Git (a common occurrence!).

**Pull request merged**